alts/src/main/java/io/grpc/alts/internal/AltsProtocolNegotiator.java

```diff
@@ -68,20 +80,28 @@ public Handler newHandler(GrpcHttp2ConnectionHandler grpcHandler) {
       @Override
       public void close() {
         logger.finest("ALTS Client ProtocolNegotiator Closed");
-        // TODO(jiangtaoli2016): release resources
+        handshakerChannelPool.returnObject(handshakerChannel);
```

Comment:
This could release the object before all the Handlers have finished using it.

Reply:
If we don't return the channel here, where shall we return the channel?

Reply:
I was thinking we would return it here and in the handshakers.

However, I checked and this close is called on termination, which means it is guaranteed no handshakes are still occurring. So there's nothing needed to be changed here.